tests/test_project.py

```diff
@@ -171,9 +172,158 @@ def test_generate_no_handlers(project):
     mock_plugin = MagicMock(spec=["generate"])
     with patch.object(project, "_plugin", mock_plugin):
         project.generate()
+        project.generate_docs()
     mock_plugin.generate.assert_called_once_with(project)
 
 
+def test_generate_with_docs(project, tmpdir_factory):
+    project.schema = resource_json(
+        __name__, "data/schema/valid/valid_type_complex.json"
+    )
+    project.type_name = "AWS::Color::Red"
+    # tmpdir conflicts with other tests, make a unique one
+    project.root = tmpdir_factory.mktemp("generate_with_docs")
+    mock_plugin = MagicMock(spec=["generate"])
+    with patch.object(project, "_plugin", mock_plugin):
+        project.generate()
+        project.generate_docs()
+    mock_plugin.generate.assert_called_once_with(project)
+
+    docs_dir = Path(project.root / "docs")
+    readme_file = Path(project.root / "docs" / "README.md")
```

Comment:
don't think the conversion is needed? `project.root` should be a `Path` already

Reply:
Changed per your suggestion.